sdk/nodejs/x/automation/stack.ts

```diff
@@ -104,6 +111,39 @@ export class Stack {
                 throw new Error(`unexpected Stack creation mode: ${mode}`);
         }
     }
+    // Try for up to 10s to start tailing the file, invoking the callback once per line. Returns
+    // a promise for a callback to invoke to stop tailing the file.
+    private async readLines(path: string, callback: (event: EngineEvent) => void): Promise<TailFile> {
+        let n = 0;
+        let fileExists = fs.existsSync(path);
+        while (!fileExists) {
+            if (n > 100) {
+                throw new Error("Log file never found.");
+            }
+            await delay(100);
+            fileExists = fs.existsSync(path);
+            n++;
+        }
+        const eventLogTail = new TailFile(path, { startPos: 0 });
+        eventLogTail
+            .on("tail_error", (err) => {
+                console.error("TailFile had an error!", err);
+                throw err;
+            })
+            .start()
+            .catch(err => {
+                throw new Error(`Cannot start, does the file exist? ${err}`);
+            });
+
+        eventLogTail
+            .pipe(split2())
+            .on("data", line => {
+                const event = JSON.parse(line);
```

Comment:
Very minor - but might be worth a type annotation here to make clear what type we are expecting to get back from the JSON.parse.